### PR TITLE
Fix scope breaking change of request/response interception. (#241)

### PR DESCRIPTION
**Pull request by Jack12816**
See: https://github.com/oesmith/puffing-billy/issues/241#issuecomment-385370968

### CODE REVIEW

**Comment by Jack12816** [comment]
Excuse me? Don't know what you mean.

**Comment by Jack12816** [comment]
Ah, you mean `subject.extend(RSpec::Matchers)`. Mhh lets see.

**Comment by Jack12816** [comment]
Yes, this is fine without due to the calling scope. I removed these calls. Thanks.

File: lib/billy.rb [hunk 1]
```diff
@@ -28,4 +28,18 @@ def self.proxy
...
     @certificate_authority ||= Billy::Authority.new
   end
+
+  # This global shortcut can be used inside of request stubs. You can modify
+  # the request beforehand and/or modify the actual response which is passed
+  # back by this method. But you can also implement a custom proxy passing
+  # method if you like to. This is just a shortcut.
+  def self.pass_request(params, headers, body, url, method)
+      handler = proxy.request_handler.handlers[:proxy]
+      response = handler.handle_request(method, url, headers, body)
```
Comment (AlanFoster) [inline lib/billy.rb]: nab: It's interesting to see the difference between internet/external APIs here.

File: lib/billy.rb [hunk 2]
```diff
@@ -28,4 +28,18 @@ def self.proxy
   def self.certificate_authority
     @certificate_authority ||= Billy::Authority.new
   end
+
+  # This global shortcut can be used inside of request stubs. You can modify
+  # the request beforehand and/or modify the actual response which is passed
+  # back by this method. But you can also implement a custom proxy passing
+  # method if you like to. This is just a shortcut.
+  def self.pass_request(params, headers, body, url, method)
```
Comment (AlanFoster) [inline lib/billy.rb]: Just to confirm, is `params` a dead variable here?

File: lib/billy.rb (hunk 1 above)
Comment (Jack12816) [reply lib/billy.rb]: Yes because of:

```ruby
proxy.stub('http://example.com/').and_return(Proc.new { |*args|
  response = pass_request(*args)
end
```

Comment (Jack12816) [reply lib/billy.rb]: Makes it easier to use here without introduce a breaking change. Its more likely to intercept the response and modify it as to tweak the request. It was fine in first place, no need to debate on it in a fixup.

**Comment by Jack12816** [comment]
@ronwsmith Ping?

**Pull request merged**